sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/export/RegisteredReader.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.export;
+
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a {@link MetricReader} registered with {@link SdkMeterProvider}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class RegisteredReader {
```

Comment:
A simple wrapper of `MetricReader` which is assigned a UUID to allows internal code to differentiate readers. 

Later, we may choose to use this class to track when a reader has last collected, which would assist in solving #4400.

Reply:
As an FYI - you could continue to use `CollectionHandle` if you wanted, which is GUID (from the standpoint of the Metrics SDK), and performs similarly to using an integer ID.

They may be less needed given the overall scope here.

Reply:
If I understand correctly, `CollectionHandle` and `CollectionInfo` are both somewhat involved in identifying a unique reader. I didn't see the need to have two classes for that concept, and figured renaming to `RegisteredReader` was more representative of what the class does. That is, it represents a unique registered reader. Internal components can rely on `hashCode()` and `equals()` that are unique among readers, and any meta data that needs to be stored with the reader can be associated with the `RegisteredReader`.